### ISSUE TITLE
Exception with Date type property

### ISSUE THREAD

**Issue by truongbb** [halab4dev/mongo2j#1]
I 've tried this library for the first time and here my POJO class:

```java
public class PointLog {

    @BsonId
    private String id;

    @BsonProperty("user_id")
    private String userId;
    
    private int point;
    
    private Date time;
}
```

And here's my main code:

```java
PointLog pointLog = new PointLog();
pointLog.setUserId("123567");
pointLog.setPoint(1000);
pointLog.setTime(new Date());

Mapper mapper = new DefaultMapper();

Document document = mapper.toDocument(pointLog);
System.out.println(document.toJson());
```

And then I got the exception like this:

```java
Exception in thread "main" java.lang.StackOverflowError
	at com.github.halab4dev.mongo2j.util.ClassUtils.getSuperClassField(ClassUtils.java:101)
	at com.github.halab4dev.mongo2j.util.ClassUtils.getSuperClassField(ClassUtils.java:103)
	at com.github.halab4dev.mongo2j.util.ClassUtils.getSuperClassField(ClassUtils.java:103)
	at com.github.halab4dev.mongo2j.util.ClassUtils.getSuperClassField(ClassUtils.java:103)
	at com.github.halab4dev.mongo2j.util.ClassUtils.getSuperClassField(ClassUtils.java:103)
	at com.github.halab4dev.mongo2j.mapper.BsonSerializer.toDocument(BsonSerializer.java:29)
	at com.github.halab4dev.mongo2j.mapper.BsonSerializer.addDocumentField(BsonSerializer.java:77)
```

Please fix this problem soon!

**Comment by halab4dev** [comment]
@truongbb I've fixed this bug on 2.0.1 version. Check it out.

**Comment by truongbb** [comment]
@halab4dev I've tried again and it seems to be good! Thank you.
